carma_wm_ctrl/src/WMBroadcaster.cpp

```diff
@@ -151,32 +152,47 @@ lanelet::ConstLaneletOrAreas WMBroadcaster::getAffectedLaneletOrAreas(const cav_
   }
 
   // Logic to detect which part is affected
-  double lane_max_width = 4;
-  
+
   std::unordered_set<lanelet::Lanelet> affected_lanelets;
   for (int idx = 0; idx < gf_pts.size(); idx ++)
   {
     std::unordered_set<lanelet::Lanelet> possible_lanelets;
-    // get nearest few nearest llts within lane_max_width
+    // get nearest few nearest llts within max_lane_width_
     // which actually house this geofence_point
     auto searchFunc = [&](const lanelet::BoundingBox2d& lltBox, const lanelet::Lanelet& llt) 
     {
-      bool should_stop = boost::geometry::distance(gf_pts[idx].basicPoint2d(), llt.polygon2d()) > lane_max_width;
-      if (!should_stop && boost::geometry::within(gf_pts[idx].basicPoint2d(), llt.polygon2d()))
+      bool should_stop_searching = boost::geometry::distance(gf_pts[idx].basicPoint2d(), llt.polygon2d()) > max_lane_width_;
+      if (!should_stop_searching && boost::geometry::within(gf_pts[idx].basicPoint2d(), llt.polygon2d()))
       {
         possible_lanelets.insert(llt);
       }
-      return should_stop;
+      return should_stop_searching;
     };
-   
+
+    // this call updates possible_lanelets
     current_map_->laneletLayer.nearestUntil(gf_pts[idx], searchFunc);
 
+    // we utilize routes to filter llts that are overlapping but not connected
+    lanelet::traffic_rules::TrafficRulesUPtr traffic_rules_car = lanelet::traffic_rules::TrafficRulesFactory::create(
+      lanelet::traffic_rules::CarmaUSTrafficRules::Location, lanelet::Participants::VehicleCar);
+    lanelet::routing::RoutingGraphUPtr map_graph = lanelet::routing::RoutingGraph::build(*current_map_, *traffic_rules_car);
```

Comment:
Can you confirm this routing graph will only be built once per geofence? I don't want to have it happen more than that. In the future we may want to find a way to optimize it further but we will leave that for future work.

Reply:
Yeah. This logic is in getAffectedLaneletOrAreas function which is only called when geofence_msg is being translated into Geofence object that happens once in GeofenceCallback.